env/expand.go

```diff
@@ -0,0 +1,195 @@
+package env
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/envman/models"
+)
+
+// Action is a possible action changing an environment variable
+type Action int
+
+const (
+	// InvalidAction represents an unexpected state
+	InvalidAction Action = iota + 1
+	// SetAction is an environment variable assignement, like os.Setenv
+	SetAction
+	// UnsetAction is an action to clear (if existing) an environment variable, like os.Unsetenv
+	UnsetAction
+	// SkipAction means that no action is performed (usually for an env with an empty value)
+	SkipAction
+)
+
+// Command describes an action performed on an envrionment variable
+type Command struct {
+	Action   Action
+	Variable Variable
+}
+
+// Variable is an environment variable
+type Variable struct {
+	Key   string
+	Value string
+	// IsSensitive is true if variable is marked (optionally) sensitive initally (for example a sensitive input or a secret),
+	// or recursively references any variable marked as sensitive.
+	// The goal is to keep track of any references secrets, so these can be redacted easily.
+	IsSensitive bool
```

Comment:
is IsSensitive used?

Other than that I am not sure why is it declared for Variables, is_sensitive affects only the Bitrise UI, we do not need to follow sensitive input references.

Later, when we will redact secrets, we will have a list of secret values and each env's value matching to any secrets value needs to be redacted.

Reply:
Was thinking about is_sensitive inputs that have not a secret input set, or they are directly given as a string.
But can be removed if you say that the current manual whitelist approach is enough to filter out is_sensitive inputs.